scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/DataFileService.java

```
@@ -0,0 +1,89 @@
+ */
+
+@Transactional
+@Service
+public class DataFileService implements IDataFileService {
+
+    private final DataFileRepository dataFileRepository;
+    private final DatasetRepository datasetRepository;
+    private final IFileSystemService iFileSystemService;
+
+    @Autowired
+    public DataFileService(DataFileRepository dataFileRepository,
+                           DatasetRepository datasetRepository,
+                           IFileSystemService iFileSystemService) {
+        this.dataFileRepository = dataFileRepository;
+        this.datasetRepository = datasetRepository;
+        this.iFileSystemService = iFileSystemService;
+    }
+
+    /**
+     * Creates a file on the filesystem and stores the DataFile
+     * in the database. A Dataset has many DataFiles and a DataFile belongs to one Dataset.
+     *
+     * @param bytes     The byte array representation of a MultipartFile
+     * @param datasetId The primary key of the Dataset as a String.
+     * @param username  The user / owner of the Dataset
+     * @param filename  The name of the file
+     * @param type      The file type
+     */
+    @Override
+    public void createFile(byte[] bytes, String datasetId, String username, String filename, String type) {
+        try {
+            iFileSystemService.createFileInDirectory(username, datasetId, filename, bytes);
+        } catch (DataFileStorageException e) {
+            throw new DataFileStorageException("File could not be saved in the file system", e);
+        }
+
+        DataFile newFile = new DataFile();
+        newFile.setName(filename);
+        newFile.setType(type);
+        newFile.setCreatedAt(OffsetDateTime.now());
+        Optional<Dataset> dataset;
+        try {
+            dataset = datasetRepository.findById(Long.parseLong(datasetId));
+        } catch (NumberFormatException e){
+            throw new DataFileStorageException("Dataset Id could not be parsed to a long");
```

don't swallow the exception, it needs to be passed into the new exception to generate the correct stack trace.